### PR TITLE
fix type error in manifest asset bundle

### LINKED ISSUE #42723: Flutter app gets stuck at splash screen
## Steps to Reproduce

I have stumbled on issue that the application gets stuck at the splash screen. No errors, no exceptions, nothing.

Screenshot 2019-10-15 at 10 48 59

You can see that the "starting app..." is printed but seems to get stuck in "GeneratedPluginRegistrant.register", although as the logs show, the flutter debugger is started.

**Target Platform:** iOS
**Target OS version/browser:** iOS 12.2
**Devices:** Simulator iPhone X

Has anyone encountered such an odd behaviour? Ideas?

## Dependenices

```

dependencies:
  flutter:
    sdk: flutter

  flutter_localizations:
    sdk: flutter

  flutter_cupertino_localizations: ^1.0.1

  provider: ^3.0.0+1
  shared_preferences: ^0.5.3
  json_annotation: ^3.0.0
  dio: 2.1.0
  package_info: ^0.4.0+3
  device_info: ^0.4.0+2
  sqflite: ^1.1.0
  path_provider: ^1.1.0
  mvc_pattern: ^3.4.3
  equatable: ^0.5.1
  barcode_scan:
    git:
      url: git://github.com/apptreesoftware/flutter_barcode_reader
      ref: 39ecc57bc0eef7b0465c891d900d2fb1f34a2390
  geolocator: "^5.1.1"
  toast: ^0.1.3
  cupertino_icons: ^0.1.2
  url_launcher: ^5.1.0
  logger: ^0.7.0+2
  flare_flutter: ^1.5.8
  flutter_opentok:
    git:
      url: git://github.com/ninjasolutions/flutter_opentok
      ref: master
  flutter_pusher: ^1.0.0
  audioplayers: ^0.13.2
```
## Logs
Running flutter with verbose output, no exception occurs. Here are the last lines:
```

[   +6 ms] executing: /usr/bin/xcrun simctl launch 6D7AD1A4-5621-4A9C-AA4F-1DE5D7B8FFB4 ninja.indoor --enable-dart-profiling --enable-checked-mode --verify-entry-points --observatory-port=0
[ +388 ms] ninja.indoor: 1552
[        ] Waiting for observatory port to be available...
[+2197 ms] [DEVICE LOG] Timestamp                       (process)[PID]
[        ] [DEVICE LOG] 2019-10-15 10:57:30.062878+0300  localhost Runner[1552]: (libAccessibility.dylib) [com.apple.Accessibility:AccessibilitySupport] Retrieving resting unlock: 0
[ +151 ms] [DEVICE LOG] 2019-10-15 10:57:30.216077+0300  localhost Runner[1552]: (Flutter) flutter: Observatory listening on http://127.0.0.1:61396/4wvsrMTFLus=/
[   +3 ms] Observatory URL on device: http://127.0.0.1:61396/4wvsrMTFLus=/
[   +3 ms] Connecting to service protocol: http://127.0.0.1:61396/4wvsrMTFLus=/
[ +146 ms] Successfully connected to service protocol: http://127.0.0.1:61396/4wvsrMTFLus=/
[   +4 ms] Sending to VM service: getVM({})
[   +6 ms] Result: {type: VM, name: vm, architectureBits: 64, hostCPU: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz, operatingSystem: ios, targetCPU: x64, version: 2.6.0-dev.7.0.flutter-70a7ef3f58 (Mon Oct 14 18:25:56 2019 +0000) on "ios_x64", _profilerMode: VM, _nativ...
[  +10 ms] Sending to VM service: getIsolate({isolateId: isolates/208097458219451})
[   +9 ms] Sending to VM service: _flutter.listViews({})
[   +3 ms] Result: {type: Isolate, id: isolates/208097458219451, name: main, number: 208097458219451, _originNumber: 208097458219451, startTime: 1571126250128, _heaps: {new: {type: HeapSpace, name: new, vmName: Scavenger, collections: 0, avgCollectionPeriodMillis: 0...
[   +4 ms] Result: {type: FlutterViewList, views: [{type: FlutterView, id: _flutterView/0x7fbdfc802a20, isolate: {type: @Isolate, fixedId: true, id: isolates/208097458219451, name: main.dart$main-208097458219451, number: 208097458219451}}]}
[   +8 ms] DevFS: Creating new filesystem on the device (null)
[   +1 ms] Sending to VM service: _createDevFS({fsName: host_flutter})
[  +37 ms] Result: {type: FileSystem, name: host_flutter, uri: file:///Users/genertorg/Library/Developer/CoreSimulator/Devices/6D7AD1A4-5621-4A9C-AA4F-1DE5D7B8FFB4/data/Containers/Data/Application/07F9FBB7-E789-419F-BA27-5DBEEB1E46FC/tmp/host_flutterEOFkgK/host_flut...
[        ] DevFS: Created new filesystem on the device (file:///Users/genertorg/Library/Developer/CoreSimulator/Devices/6D7AD1A4-5621-4A9C-AA4F-1DE5D7B8FFB4/data/Containers/Data/Application/07F9FBB7-E789-419F-BA27-5DBEEB1E46FC/tmp/host_flutterEOFkgK/host_flutter/)
[   +4 ms] Updating assets
[ +458 ms] Syncing files to device iPhone X...
[   +2 ms] Scanning asset files
[  +29 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[  +12 ms] /Users/genertorg/flutter/bin/cache/dart-sdk/bin/dart /Users/genertorg/flutter/bin/cache/artifacts/engine/darwin-x64/frontend_server.dart.snapshot --sdk-root /Users/genertorg/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --strong
--target=flutter -Ddart.developer.causal_async_stacks=true --output-dill /var/folders/tb/3yd9xyqj5cn53csl1k17tmgr0000gn/T/flutter_tool.K1sXdn/app.dill --packages /Users/genertorg/indoor.ninja/host_flutter/.packages -Ddart.vm.profile=false -Ddart.vm.product=false
--bytecode-options=source-positions,local-var-info,debugger-stops,instance-field-initializers,keep-unreachable-code,avoid-closure-call-instructions --enable-asserts --filesystem-scheme org-dartlang-root
[  +11 ms] <- compile package:flutter_ninja/main_live.dart
[+10493 ms] Updating files
[ +946 ms] DevFS: Sync finished
[        ] Syncing files to device iPhone X... (completed in 11,497ms, longer than expected)
[   +1 ms] Synced 5.9MB.
[   +2 ms] Sending to VM service: _flutter.listViews({})
[   +2 ms] Result: {type: FlutterViewList, views: [{type: FlutterView, id: _flutterView/0x7fbdfc802a20, isolate: {type: @Isolate, fixedId: true, id: isolates/208097458219451, name: main.dart$main-208097458219451, number: 208097458219451}}]}
[   +1 ms] <- accept
[        ] Connected to _flutterView/0x7fbdfc802a20.
[   +2 ms] 🔥  To hot reload changes while running, press "r". To hot restart (and rebuild state), press "R".
[   +1 ms] An Observatory debugger and profiler on iPhone X is available at: http://127.0.0.1:61396/4wvsrMTFLus=/
[        ] For a more detailed help message, press "h". To detach, press "d"; to quit, press "q".
```

```
Analyzing host_flutter...

   info • Avoid annotating types for function expression parameters • lib/generated/i18n.dart:628:13 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/generated/i18n.dart:628:35 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/generated/i18n.dart:638:13 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/generated/i18n.dart:638:28 • avoid_types_on_closure_parameters
   info • Avoid async functions that return void • lib/main_dev.dart:5:6 • avoid_void_async
   info • Avoid async functions that return void • lib/main_live.dart:5:6 • avoid_void_async
   info • Avoid async functions that return void • lib/main_mock.dart:5:6 • avoid_void_async
   info • Avoid async functions that return void • lib/main_prelive.dart:5:6 • avoid_void_async
   info • This class (or a class which this class inherits from) is marked as '@immutable', but one or more of its instance fields are not final: MyApp.fcmDataCallback •
          lib/src/app.dart:16:7 • must_be_immutable
   info • Avoid annotating types for function expression parameters • lib/src/common/platform_text.dart:36:10 • avoid_types_on_closure_parameters
   info • Avoid empty catch blocks • lib/src/data/api_helper.dart:43:19 • empty_catches
   info • Use isEmpty instead of length • lib/src/data/database/dao/account_dao.dart:48:9 • prefer_is_empty
   info • Use isEmpty instead of length • lib/src/data/database/dao/recent_dao.dart:47:9 • prefer_is_empty
   info • Use isEmpty instead of length • lib/src/data/database/dao/recent_dao.dart:64:9 • prefer_is_empty
   info • Use isEmpty instead of length • lib/src/data/database/dao/recent_dao.dart:81:9 • prefer_is_empty
   info • Use isEmpty instead of length • lib/src/data/database/dao/recent_dao.dart:107:9 • prefer_is_empty
   info • Avoid empty catch blocks • lib/src/data/device_app_info.dart:40:17 • empty_catches
   info • Avoid empty catch blocks • lib/src/data/device_app_info.dart:44:17 • empty_catches
   info • Unused import: 'dart:io' • lib/src/data/fcm/fcm_helper.dart:1:8 • unused_import
   info • The value of the local variable 'data' isn't used • lib/src/data/fcm/fcm_helper.dart:17:13 • unused_local_variable
   info • The value of the local variable 'notification' isn't used • lib/src/data/fcm/fcm_helper.dart:22:13 • unused_local_variable
   info • Unused import: 'package:equatable/equatable.dart' • lib/src/data/fcm/push_type.dart:1:8 • unused_import
   info • The declaration '_handleMethodCall' isn't referenced • lib/src/data/native_connection_helper.dart:32:19 • unused_element
   info • Use isNotEmpty instead of length • lib/src/data/requestHelper/_request_queue.dart:149:13 • prefer_is_empty
   info • Prefer using /// for doc comments • lib/src/model/lift_call_state.dart:29:3 • slash_for_doc_comments
   info • Prefer using /// for doc comments • lib/src/model/lift_call_state.dart:34:3 • slash_for_doc_comments
   info • Prefer using /// for doc comments • lib/src/model/lift_call_state.dart:43:3 • slash_for_doc_comments
   info • Prefer using /// for doc comments • lib/src/model/lift_call_state.dart:52:3 • slash_for_doc_comments
   info • Avoid annotating types for function expression parameters • lib/src/screens/dialog/update_dialog.dart:46:19 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/firebase_messaging_example.dart:19:24 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/firebase_messaging_example.dart:19:41 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/lift/elevator_call_state/elevator_call_state_controller.dart:97:21 •
          avoid_types_on_closure_parameters
   info • Unused import: 'package:flutter_ninja/app_state_container.dart' • lib/src/screens/lift/floor/floor_selection_button.dart:2:8 • unused_import
   info • Unused import: 'package:flutter_ninja/src/bloc/user_bloc.dart' • lib/src/screens/lift/lift_view.dart:6:8 • unused_import
   info • Avoid annotating types for function expression parameters • lib/src/screens/lift/lift_view.dart:195:23 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/locks/area_locks_view.dart:251:28 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/locks/lock_item.dart:60:20 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/locks/locks_view.dart:95:28 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/login/login_controller.dart:44:57 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/login/login_controller.dart:62:21 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/login/login_controller.dart:87:52 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/login/login_controller.dart:96:21 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/main/drawer/about/about_view.dart:105:37 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/main/drawer/about/about_view.dart:111:57 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/main/drawer/about/about_view.dart:117:38 • avoid_types_on_closure_parameters
   info • Unused import: 'package:flutter_ninja/src/screens/main/drawer/about/about_view.dart' • lib/src/screens/main/drawer/drawer.dart:4:8 • unused_import
   info • Avoid annotating types for function expression parameters • lib/src/screens/main/drawer/settings_item.dart:131:27 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/main/main_controller.dart:188:57 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/main/main_controller.dart:203:52 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/main/main_controller.dart:213:50 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/main/main_controller.dart:221:21 • avoid_types_on_closure_parameters
   info • Use `;` instead of `{}` for empty constructor bodies • lib/src/screens/parcel/parcel_controller.dart:30:55 • empty_constructor_bodies
   info • Avoid annotating types for function expression parameters • lib/src/screens/visitor/locks_action/locks_action_view.dart:97:28 • avoid_types_on_closure_parameters
   info • Avoid using braces in interpolation when not needed • lib/src/screens/visitor/pusher_controller.dart:174:43 • unnecessary_brace_in_string_interps
   info • Avoid annotating types for function expression parameters • lib/src/screens/visitor/visitor_controller.dart:80:24 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/visitor/visitor_controller.dart:80:41 • avoid_types_on_closure_parameters
   info • Avoid annotating types for function expression parameters • lib/src/screens/visitor/visitor_controller.dart:281:21 • avoid_types_on_closure_parameters
   info • The value of the field '_apiCaller' isn't used • lib/src/screens/visitor/visitor_view.dart:555:19 • unused_field
   info • `Future` results in `async` function bodies must be `await`ed or marked `unawaited` using `package:pedantic` • lib/src/screens/visitor/visitor_view.dart:575:5 •
          unawaited_futures
   info • Unnecessary new keyword • lib/src/util/validate.dart:10:21 • unnecessary_new
   info • This class (or a class which this class inherits from) is marked as '@immutable', but one or more of its instance fields are not final: WidgetMat.widget •
          test/widget_materializer.dart:4:7 • must_be_immutable
```

```
[✓] Flutter (Channel master, v1.10.15-pre.121, on Mac OS X 10.15 19A583, locale en-GB)
    • Flutter version 1.10.15-pre.121 at /Users/genertorg/flutter
    • Framework revision b857632306 (43 minutes ago), 2019-10-15 00:07:32 -0700
    • Engine revision 540fc977bb
    • Dart version 2.6.0 (build 2.6.0-dev.7.0 70a7ef3f58)


[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/genertorg/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.3, Build version 10G8
    • CocoaPods version 1.8.3

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 40.0.2
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.39.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.5.1

[✓] Connected device (1 available)
    • iPhone X • 6D7AD1A4-5621-4A9C-AA4F-1DE5D7B8FFB4 • ios • com.apple.CoreSimulator.SimRuntime.iOS-12-2 (simulator)

• No issues found!
```

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Fixes https://github.com/flutter/flutter/issues/42723

Adds regression test.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/asset_bundle_test.dart [hunk 1]
```diff
@@ -148,6 +148,33 @@ name: example''')
       FileSystem: () => testFileSystem,
       ProcessManager: () => FakeProcessManager(<FakeCommand>[]),
     });
+
+    // https://github.com/flutter/flutter/issues/42723
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/asset_bundle_test.dart]: Do we usually add issues URLs for regression tests?  I feel like these get out of date when the tests change, vs just looking at the PR via source control.

**Review by jmagman — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/general.shard/asset_bundle_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/asset_bundle_test.dart]: I usually do - in this case I also added a comment to make it clearer what I am testing

File: packages/flutter_tools/lib/src/asset.dart [hunk 2]
```diff
@@ -94,7 +94,7 @@ class _ManifestAssetBundle implements AssetBundle {
       if (!directory.existsSync()) {
         return true; // directory was deleted.
       }
-      for (File file in directory.listSync()) {
+      for (File file in directory.listSync().whereType<File>()) {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/asset.dart]: Why is this not a recursive listing?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/asset.dart]: Wildcard assets aren't recursive

Comment (zanderso) [reply packages/flutter_tools/lib/src/asset.dart]: Ahh, okay. Might be worth a comment here when you get a chance.

**Comment by natanloterio** [comment]
Hi guys, I'm getting the same error. Is there any workaround?

**Comment by jonahwilliams** [comment]
@natanloterio what version of flutter are you on? unfortunately we haven't been able to do a dev release in a while but if you're on master an update will work.

**Comment by natanloterio** [comment]
@jonahwilliams , I figured out that the problem was with the Master Channel's version. When I switched back to the Stable Channel, the hot-load started to work back again.

Sorry, it was my fault.

Thank you!

**Pull request merged**